src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -19,344 +15,360 @@
+DeviceInformation Plugin::parse_batch_device(const std::string& device_with_batch) {
+    auto openingBracket = device_with_batch.find_first_of('(');
+    auto closingBracket = device_with_batch.find_first_of(')', openingBracket);
+    auto deviceName = device_with_batch.substr(0, openingBracket);
 
     int batch = 0;
     if (closingBracket != std::string::npos && openingBracket < closingBracket) {
-        batch = std::stol(d.substr(openingBracket + 1, closingBracket - 1));
+        batch = std::stol(device_with_batch.substr(openingBracket + 1, closingBracket - 1));
 
         if (batch <= 0) {
-            IE_THROW() << "Batch value for '" << deviceName << "' must be > 0, while " << batch << "is passed";
+            OPENVINO_THROW("Batch value for '", deviceName, "' must be > 0, while ", batch, "is passed");
         }
     }
-    return {deviceName, {{}}, batch};
+    return {deviceName, {{}}, static_cast<uint32_t>(batch)};
 }
 
-DeviceInformation Plugin::ParseMetaDevice(const std::string& devicesBatchCfg,
-                                          const std::map<std::string, std::string>& user_config) const {
-    auto metaDevice = ParseBatchDevice(devicesBatchCfg);
-    metaDevice.config = GetCore()->GetSupportedConfig(metaDevice.device_name, user_config);
-
+DeviceInformation Plugin::parse_meta_device(const std::string& devices_batch_config,
+                                            const ov::AnyMap& user_config) const {
+    auto meta_device = parse_batch_device(devices_batch_config);
+    meta_device.device_config = get_core()->get_supported_property(meta_device.device_name, user_config);
     // check that no irrelevant config-keys left
     for (const auto& k : user_config) {
         const auto& name = k.first;
-        if (metaDevice.config.find(name) == metaDevice.config.end() &&
+        if (meta_device.device_config.find(name) == meta_device.device_config.end() &&
             !ov::util::contains(supported_configKeys, name)) {
-            IE_THROW() << "Unsupported config key: " << name;
+            OPENVINO_THROW("Unsupported config key: ", name);
         }
     }
-    return metaDevice;
+    return meta_device;
 }
 
-InferenceEngine::RemoteContext::Ptr Plugin::CreateContext(const InferenceEngine::ParamMap& remote_properties) {
-    auto cfg = remote_properties;
-    auto it = cfg.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
-    if (it == cfg.end())
-        it = cfg.find(ov::device::priorities.name());
-    if (it == cfg.end())
-        IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
-
-    auto val = it->second.as<std::string>();
-    auto core = GetCore();
-    if (!core)
-        return nullptr;
-    auto metaDevice = ParseMetaDevice(val, std::map<std::string, std::string>());
-    cfg.erase(it);
-    return core->CreateContext(metaDevice.device_name, cfg);
+ov::SoPtr<ov::IRemoteContext> Plugin::create_context(const ov::AnyMap& remote_properties) const {
+    OPENVINO_NOT_IMPLEMENTED;
```

Comment:
Why did you remove this functionality???? Previously we had an opportunity to create context